### ISSUE TITLE
Rendering LocalForm more than once with different initialStates

### ISSUE THREAD

**Issue by fraxachun** [davidkpiano/react-redux-form#857]
### The Problem
When I use a single LocalForm Class which is rendered more than once with different initialState it won't update the Controls accordingly

### Steps to Reproduce
In CodePen click on "foo" then on "bar"

### Expected Behavior
When clicking on "bar" the Control.text should update to "bar"

### Actual Behavior
The input stays on "foo" (not only the value, the whole Control.text is linking to the old model)

### Reproducible Code Example
[CodePen](https://codepen.io/anon/pen/ZyymoP)

**Comment by cipater** [comment]
@fraxachun I just ran into this as well. I'd like to see a better API for this for LocalForm... but it seems the current way to do this is:

```javascript
<LocalForm getDispatch={ dispatch => this.formDispatch = dispatch } initialState={ state } />

// when you want to update the initialState
this.formDispatch(actions.change('local', nextState))
```

**Comment by fraxachun** [comment]
Ok, in the end this turns out to be a logical solution and even gives more control about the forms behaviour. Thanks!

**Comment by davidkpiano** [comment]
> I just ran into this as well. I'd like to see a better API for this for LocalForm... but it seems the current way to do this is:

@cipater What's your ideal API for this?

**Comment by cipater** [comment]
I'm torn on this. On the one hand, I'd expect the behavior @fraxachun mentions—if I pass in new data, I expect that new data to be rendered. After all, Flux/redux/et al are patterns/tools designed to encourage removing state from components. And from the outside looking in, `LocalForm` is somewhat ironically using redux to do just the opposite.

On the other, while `LocalForm` has gone a long way to abstract away most of the complexities of setting up / using redux, this is nonetheless a redux-based project. And if you want to move beyond the basics while still using `LocalForm`, the use of `getDispatch` seems unavoidable. 

As such, I suspect further abstractions are outside the scope of this project. However, I'd ideally like to see that abstraction pushed as far as is reasonable. i.e. with `initialState`, maybe have `LocalForm` do something like `actions.change(nextProps.model, nextProps.initalState)` inside `componentWillReceiveProps`. Or on some other prop if using `initialState` in this way a) is confusing given its name, or b) might break existing code.

**Comment by davidkpiano** [comment]
I'll keep that in consideration!

**Comment by KubaChoice** [comment]
Hi, It will be helpful (in my case) if initialState could take a Promise. What do you think?

**Comment by davidkpiano** [comment]
@KubaChoice Good idea, open a new issue (or better yet, a pull request! it's not too hard) for that please 🙏
